client/src/components/OptionsetField/OptionField.js

```diff
@@ -76,25 +70,14 @@ class OptionField extends Component {
       ? this.props.leftTitle
       : this.props.title;
 
-    // default and fallback to a Radio button
-    let Option = null;
-
-    switch (this.props.type) {
-      case 'checkbox':
-        Option = Checkbox;
-        break;
-      case 'radio':
-        Option = Radio;
-        break;
-      default:
-        throw new Error(`Invalid OptionField type: ${this.props.type}`);
-    }
-
-    const label = (typeof labelText === 'string')
-      ? { react: <span>{labelText}</span> }
-      : labelText;
-
-    return castStringToElement(Option, label, this.getInputProps());
+    return (
+      <FormGroup check>
+        <Label check>
+          <Input {...this.getInputProps()} />
+          {labelText}
```

Comment:
just realised the `castStringToElement()` call missing for this label, which is important for if the PHP returns an html object.